chef/cookbooks/keystone/libraries/helpers.rb

```diff
@@ -36,7 +36,11 @@ def self.keystone_settings(current_node, cookbook_name)
 
       @keystone_settings ||= Hash.new
       @keystone_settings[cookbook_name] = {
-        "api_version" => node[:keystone][:api][:version].sub(/^v/, ""),
+        "api_version" => node[:keystone][:api][:version],
+        # This is somehwat ugly but the Juno keystonemiddleware expects the
+        # version to be a "v3.0" for the v3 API instead of the "v3" or "3" that
+        # is used everywhere else.
+        "api_version_for_middleware" => "v#{node[:keystone][:api][:version].to_f}",
```

Comment:
is .to_f always generating a one digit suffix? I suggest

```ruby
 "v%.1f" % node[:keystone][:api][:version]
```

which makes this more explicit IMHO